### ISSUE TITLE
[DataTable] Overflow of DataTable does not work properly if there are tooltips inside

### ISSUE THREAD

**Issue by jakubfaliszewski** [carbon-design-system/carbon#6728]
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

When bottom row of DataTable has tooltip inside (`Button` with tooltip in demo) whole table becomes scrollable in Y axis and there's no possibility to see tooltip until user scrolls down a little 
> Is this issue related to a specific component?

`DataTable`>`.bx--data-table-content`

> What did you expect to happen? What happened instead? What would you like to
> see changed?

Tooltip should be visible without scrolling - like when there was no `overflow-x: auto` for `.bx--data-table-content`

> What browser are you working in?

Chrome 84

## Steps to reproduce the issue

1. Create DataTable
2. Add some element inside last row that sticks out from the row from the bottom

> Please create a reduced test case in CodeSandbox

Demo with Button with tooltip: https://codesandbox.io/s/relaxed-monad-tfc1k

![demox](https://user-images.githubusercontent.com/25402419/91309218-9d725e80-e7b0-11ea-9023-5460f2a449ad.gif)

**Comment by galilmori** [comment]
Hello
We have the same problem with the datatable and tooltips in Power-iaas team.
Is there any advancements on this issue? this Bug is sometimes vey blocking for our clients who misses the information not realizing the scrollbar next to it (not to mention its esthetically very annoying for the UX designers).

Thank you for the information.

**Comment by tw15egan** [comment]
@galilmori @jakubfaliszewski as a temporary workaround, you can always set `tooltipDirection="top"` on the last item in your table. Here's an example: https://codesandbox.io/s/ancient-voice-kdnzu?file=/src/App.js

**Comment by galilmori** [comment]
@tw15egan unfortunately this workaround would only fix half  of the problem form me since we have icons close to the table border where it creates scrolls on two dimensions 
![image](https://user-images.githubusercontent.com/50196414/100203511-fc0bbd80-2f02-11eb-94d7-122e065f644b.png)

**Comment by tw15egan** [comment]
You should also be able to use the `tooltipAlignment` prop to set the carrot to the other side

**Comment by alldayalone** [comment]
I have the same issue

**Comment by justindm234** [comment]
Same issue here.

**Comment by sstrubberg** [comment]
Is this still an issue / have you tried using popover to resolve this?

**Comment by jakubfaliszewski** [comment]
Yup, here's updated example:
https://codesandbox.io/s/awesome-boyd-kyx2we?file=/index.js

Popover may work here, haven't tested it, but only as a workaround

**Comment by Kilian-Collender** [comment]
Any movement on this issue?

**Comment by tay1orjones** [comment]
@Kilian-Collender In the time since this issue was opened, Popover now has an `autoAlign` prop. It's not passed down through IconButton (but it probably should). We know there's some shortcomings with it though - notably it only detects window boundaries, not element boundaries, https://github.com/carbon-design-system/carbon/issues/13279. Depending on your usage it may not reflow the tooltip as expected.

**Comment by tw15egan** [comment]
@tay1orjones I just took a quick look at this, and I'm wondering if we could add a new prop,  kind of like `floating-menu-container` / `popoverContainerSelector`, that if passed in, would replace the coords for the [conditions check](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Popover/index.tsx#L193-L202)

```js
    const rect = popoverContainerSelector
      ? document.querySelector(popoverContainerSelector)?.getBoundingClientRect()
      : floating.current.getBoundingClientRect();
```

Haven't tested, but could be a good starting point if anyone wants to take a deeper look

**Comment by riddhybansal** [comment]
Hi there everyone this has been fixed by #14139 
Here is the [stackblitz](https://stackblitz.com/edit/github-zxv291-76fzad?file=src%2FApp.jsx) example of the same. The tooltip will align itself wherever it will get space without creating any extra scrollbars.
